src/code/flg_set.c

```diff
@@ -114,14 +114,14 @@ static FlagSetEntry sFlagEntries[] = {
     { &gSaveContext.save.saveInfo.weekEventReg[98], "week_event_reg[98]" },
     { &gSaveContext.save.saveInfo.weekEventReg[99], "week_event_reg[99]" },
 
-    { &gSaveContext.eventInf[0], "event_inf[0]" },
-    { &gSaveContext.eventInf[1], "event_inf[1]" },
-    { &gSaveContext.eventInf[2], "event_inf[2]" },
-    { &gSaveContext.eventInf[3], "event_inf[3]" },
-    { &gSaveContext.eventInf[4], "event_inf[4]" },
-    { &gSaveContext.eventInf[5], "event_inf[5]" },
-    { &gSaveContext.eventInf[6], "event_inf[6]" },
-    { &gSaveContext.eventInf[7], "event_inf[7]" },
+    { &gSaveContext.eventInf[EVENTINF_00], "event_inf[0]" },
+    { &gSaveContext.eventInf[EVENTINF_01], "event_inf[1]" },
+    { &gSaveContext.eventInf[EVENTINF_02], "event_inf[2]" },
+    { &gSaveContext.eventInf[EVENTINF_03], "event_inf[3]" },
+    { &gSaveContext.eventInf[EVENTINF_04], "event_inf[4]" },
+    { &gSaveContext.eventInf[EVENTINF_05], "event_inf[5]" },
+    { &gSaveContext.eventInf[EVENTINF_06], "event_inf[6]" },
+    { &gSaveContext.eventInf[EVENTINF_07], "event_inf[7]" },
```

Comment:
The EVENTINF macros are bitpacked macros like the weekEventReg ones, so just using those 8 macros would be wrong.
See `CHECK_EVENTINF` and similar macros

Reply:
No, I think in this case, it's correct. It's *not* looking at the bitpacked values or anything here. The `CHECK_EVENTINF` macro looks like this:
```c
#define CHECK_EVENTINF(flag) (gSaveContext.eventInf[(flag) >> 4] & (1 << ((flag) & 0xF)))
```

Which is way more than what this is doing.

Reply:
Feel like we should copy
```c
#define WEEKEVENTREG(index) (gSaveContext.save.saveInfo.weekEventReg[(index)])
#define GET_WEEKEVENTREG(index) ((void)0, WEEKEVENTREG(index))
```
but for event infos, for these instances where they do not seem to be bitpacked like normal. Mainly because EVENTINF_02 doesn't mean eventInfo[2] and so would be named differently.

Reply:
> No, I think in this case, it's correct. It's _not_ looking at the bitpacked values or anything here. The `CHECK_EVENTINF` macro looks like this:
> 
> ```c
> #define CHECK_EVENTINF(flag) (gSaveContext.eventInf[(flag) >> 4] & (1 << ((flag) & 0xF)))
> ```
> 
> Which is way more than what this is doing.

No, it isn't correct. Again, the macros are bitpacked. Look at it this way:
- Each macro consists on a 1 byte number (2 [nibbles](https://en.wikipedia.org/wiki/Nibble)) which can be seen as `0xXY`
- The `X` part of the macro is meant to be used to index the `eventInf` array
- The `Y` part is used _as a shift_ value to mask the the actual value contained on each entry of the array.

Knowing this, let's take a look at the macros you are using:
```c
#define EVENTINF_00 0x00
#define EVENTINF_01 0x01
#define EVENTINF_02 0x02
#define EVENTINF_03 0x03
#define EVENTINF_04 0x04
#define EVENTINF_05 0x05
#define EVENTINF_06 0x06
#define EVENTINF_07 0x07
```
All of them have an index value (the `X` part) as zero, so all of them are meant to be used to access the entry 0 of the array. But you are using the shift part of the macro (the `Y` part) to index the array, which is not how those macros are meant to be used.

Is it clear now?

Reply:
Yeah, this was a misunderstanding on my part. I thought each of these was used to represent one *byte* in the `eventInf` array, but that's not how it works. I'll fix it up.

Reply:
I ended up just reverting all this stuff back to raw integers. This PR is already pretty complicated, so I didn't see a lot of value in making it even more complicated. For now, all I did is comment out `EVENTINF_00` to `EVENTINF_07`, with a comment above saying the whole byte is used for dog racetrack info.